theano/gof/type.py

```diff
@@ -497,6 +497,9 @@ def c_sync(self, name, sub):
     def c_code_cache_version(self):
         return (1,)
 
+    def may_share_memory(self, other):
+        return self is other
```

Comment:
I don't think this is correct, for two reasons:
- two generic Python objects could share memory even if they are not the same object
- `may_share_memory` should be static and compare its two arguments, nor not be a static method and compare the arguments other than self.